src/App.jsx

```diff
@@ -14,45 +17,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+function resultGoodsArr(goods, { sortField, reverse }) {
+  const resultingArr = [...goods];
+
+  if (sortField) {
+    resultingArr.sort((good1, good2) => {
+      switch (sortField) {
+        case (SORT_FIELD_ALPHABET):
+          return good1.localeCompare(good2);
+        case (SORT_FIELD_LENGTH):
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    resultingArr.reverse();
+  }
+
+  return resultingArr;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
```

Comment:
please use more clear name
```suggestion
  const [sortType, setSortType] = useState('');
  const [isReversed, setIsReversed] = useState(false);
```